src/components/AutoComplete/AutoComplete.tsx

```diff
@@ -0,0 +1,196 @@
+    selectItem(null);
+    closeMenu();
+  };
+
+  const menuIsOpen = isOpen || (__internal__open ?? false);
+  const showClearIcon = (menuIsOpen && items.length > 0) || inputValue !== "";
+  const showMagnifier = !showClearIcon;
+  const loading = status === "LOADING" || __internal__loading;
+
+  const renderListItem = (record: Item) => {
+    if (ListItem) {
+      return <ListItem inputValue={inputValue} item={record} />;
+    }
+    return itemToString ? itemToString(record) : record;
+  };
+
+  return (
+    <Field
+      optional={optional}
+      disabled={disabled}
+      label={label}
+      hideLabel={hideLabel}
+      auxId={auxId}
+      helpText={helpText}
+      errors={fieldErrors}
+      testId={testId}
+    >
+      <div css={theme.autoComplete.getCSS({ targetElement: "container" })}>
+        <div {...getComboboxProps()}>
+          <InternalInput
+            {...getInputProps({ refKey: "innerRef", ref: innerRef })}
+            label={label}
+            describedBy={describedBy}
+            onFocus={onFocus}
+            onBlur={onBlur}
+            placeholder={placeholder}
+            __internal__focus={__internal__focus}
+            autoComplete="off"
+            hasSuffixButton
+            isLoading={loading}
+          />
+        </div>
+        <div css={theme.autoComplete.getCSS({ targetElement: "right" })}>
+          {loading && <LoadingIcon />}
+
+          <button
+            type="button"
+            onClick={onClear}
+            tabIndex={-1}
+            css={theme.autoComplete.getCSS({
+              targetElement: "clearIcon",
+              showClearIcon: showClearIcon,
+            })}
+          >
+            <Icon name="cross" />
+            <VisuallyHidden>Clear</VisuallyHidden>
+          </button>
+
+          {showMagnifier && (
+            <button
+              type="button"
+              {...getToggleButtonProps()}
+              css={theme.autoComplete.getCSS({ targetElement: "searchIcon" })}
+            >
+              <Icon name="search" />
+              <VisuallyHidden>Search</VisuallyHidden>
+            </button>
+          )}
+        </div>
+        <ul
+          {...getMenuProps()}
+          css={theme.autoComplete.getCSS({
+            targetElement: "ul",
+            isOpen: menuIsOpen,
+          })}
+        >
+          {menuIsOpen && (
+            <>
+              {items.map((record, index) => (
+                <li
+                  key={record.id || index}
+                  css={theme.autoComplete.getCSS({
+                    targetElement: "li",
+                    isHighlighted:
+                      highlightedIndex === index ||
+                      __internal__highlightedIndex === index,
+                  })}
+                  {...getItemProps({ index, item: record })}
+                >
+                  {renderListItem(record)}
+                </li>
+              ))}
+              {Footer && <Footer closeMenu={closeMenu} />}
```

Comment:
Given that this is not in the list, is this keyboard accessible? Can a keyboard user actually select this item without having to resort to clicking on it.

Reply:
I need to investigate keyboard for footer separately